File: kotlin/sum-of-multiples/src/main/kotlin/SumOfMultiples.kt
object SumOfMultiples {
    fun sum(multiplesOf: Set<Int>, upTo: Int): Int {
        return (1..(upTo - 1)).filter { n ->
            multiplesOf.indexOfFirst { m ->
                n % m == 0
            } > -1
        }.sum()
    }
}

File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

import Data.List (nub)

isMult :: Integer -> Integer -> Bool
isMult a b = a `mod` b == 0

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum $ filter p [1..(limit-1)]
  where p x = any (isMult x) factors
